### PR TITLE
Consistent log messages

### PR DESCRIPTION
**Pull request by adams85**
### Describe the purpose of your pull request

Revises log messages to make them consistent with other SDKs and also adds event ids for easier log event identification.

Note: The current logging abstraction (slf4j v1.7) doesn't support structured logging, so it can't be done without [updating to slf4j v2](https://stackoverflow.com/a/58710096).

### Requirement checklist (only if applicable)

- [ ] I have covered the applied changes with automated tests.
- [ ] I have executed the full automated test set against my changes.
- [ ] I have validated my changes against all supported platform versions.
- [ ] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

**Comment by adams85** [comment]
There are several`getXxx` evaluation method which swallow the exception but logs no error. For example: https://github.com/configcat/android-sdk/pull/30/files#diff-3847dfd0145ed85544942a1a6d4ec7c30eac878843ed6f19f1cf08958a7c2959L87

In these cases we should log an 1002 error as well.

File: src/main/java/com/configcat/ConfigCatClient.java [hunk 1]
```diff
@@ -531,34 +531,34 @@ private String getVariationIdFromSettingsMap(SettingResult settingResult, String
...
                 this.hooks.invokeOnFlagEvaluated(EvaluationDetails.fromError(key, null, error, userObject));
-                this.logger.error(error);
+                this.logger.error(1000, errorMessage);
                 return defaultVariationId;
             }
 
             Setting setting = settings.get(key);
             if (setting == null) {
-                String error = "Variation ID not found for key " + key + ". Here are the available keys: " + String.join(", ", settings.keySet());
+                String error = "Failed to evaluate setting '" + key + "' (the key was not found in config JSON). Returning the `defaultVariationId` parameter that you specified in your application: '" + defaultVariationId + "'. Available keys: " + String.join(", ", settings.keySet()) + ".";
```
Comment (adams85) [inline src/main/java/com/configcat/ConfigCatClient.java:542]: Available keys should be surrounded with single quotes like `'Key1', 'Key2', 'Key3'`

File: src/main/java/com/configcat/ConfigCatClient.java [hunk 2]
```diff
@@ -502,26 +502,26 @@ private <T> T getValueFromSettingsMap(Class<T> classOfT, SettingResult settingRe
...
                 this.hooks.invokeOnFlagEvaluated(EvaluationDetails.fromError(key, defaultValue, error, userObject));
-                this.logger.error(error);
+                this.logger.error(1000, errorMessage);
                 return defaultValue;
             }
 
             Setting setting = settings.get(key);
             if (setting == null) {
-                String error = "Value not found for key " + key + ". Here are the available keys: " + String.join(", ", settings.keySet());
+                String error = "Failed to evaluate setting '" + key + "' (the key was not found in config JSON). Returning the `defaultValue` parameter that you specified in your application: '" + defaultValue + "'. Available keys: " + String.join(", ", settings.keySet()) + ".";
```
Comment (adams85) [inline src/main/java/com/configcat/ConfigCatClient.java:513]: Available keys should be surrounded with single quotes like `'Key1', 'Key2', 'Key3'`

File: src/main/java/com/configcat/ConfigCatClient.java [hunk 3]
```diff
@@ -134,10 +138,11 @@ public <T> EvaluationDetails<T> getValueDetails(Class<T> classOfT, String key, U
             return this.getValueDetailsAsync(classOfT, key, user, defaultValue).get();
         } catch (InterruptedException e) {
             String error = "Thread interrupted.";
-            this.logger.error(error, e);
+            this.logger.error(0, error, e);
             Thread.currentThread().interrupt();
             return EvaluationDetails.fromError(key, defaultValue, error + ": " + e.getMessage(), user);
         } catch (Exception e) {
+            this.logger.error(2001, ConfigCatLogMessages.getSettingEvaluationFailedForOtherReason(key, "defaultValue", defaultValue), e);
```
Comment (adams85) [inline src/main/java/com/configcat/ConfigCatClient.java:145]: Wouldn't this rather be an 1002 (with `getSettingEvaluationErrorWithDefaultValue`) error like in the case of `getValue`?

File: src/main/java/com/configcat/ConfigCatLogMessages.java [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+    }
+
+    private static String convertKeysSetToFormatedString(final Set<String> availableKeys){
+        StringBuilder sb = new StringBuilder();
+        Iterator<String> it = availableKeys.iterator();
+        if(it.hasNext()) {
+            sb.append("'").append(it.next()).append("'");
+        }
+        while(it.hasNext()) {
+            sb.append(",").append("'").append(it.next()).append("'");
```
Comment (adams85) [inline src/main/java/com/configcat/ConfigCatLogMessages.java:89]:
```suggestion
            sb.append(", ").append("'").append(it.next()).append("'");
```

**Review by z4kn4fein — APPROVED** [review]

**Pull request merged**